src/compile/format.ts

```diff
@@ -121,7 +138,11 @@ export function formatCustomType({
 }) {
   field ??= fieldToFormat(fieldOrDatumDef, expr, normalizeStack);
 
-  if (isFieldDef(fieldOrDatumDef) && isBinning(fieldOrDatumDef.bin)) {
+  if (
+    field !== 'datum.value' && // For axis/legend, we can't correct know the end of the bin from `datum`
```

Comment:
It turns out that this part of the code always trigger crashes

-- see [this example](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEzjQATjRyZ289ABU1O2WQAEZQWyZJBp2QwRM460wDM265NhonTUOa40ZIogAO40yvRoAIxiAAxK8EFYMfFKykiYKKigDOqCaCqZSARQSOoQUjLyAL5KyOoA1migmACeOHCFOGw+mCGYbGyCOjhomOoMcHUgcLL+yj7BOSAAHi0gTD7jk3BKgXCCyoUAsjSCagD6neqXAOI2lvJK7Z2FAI4MSLI6WTqkISQqxoEA27k8mUK0RC4K8+g6XXQPVCIBqaKU-lkgWWoCgDAggwQADEPHCEaDUBMpmigA)